tests/auto/jobs/build.py

```diff
@@ -100,49 +111,84 @@ def remove_pr_data(job_obj, pr_repo_loc, repo_dir_str, rt_dir):
 def clone_pr_repo(job_obj, workdir):
     ''' clone the GitHub pull request repo, via command line '''
     logger = logging.getLogger('BUILD/CLONE_PR_REPO')
-    repo_name = 'ufs-community/ufs-srweather-app'
-    branch = 'develop'
-    git_url = f'https://${{ghapitoken}}@github.com/{repo_name}'
-    logger.debug(f'GIT URL: {git_url}')
+
+    app_name = 'ufs-srweather-app'
+    # These are for the new/head repo in the PR
+    new_name = job_obj.preq_dict['preq'].head.repo.name
+    new_repo = job_obj.preq_dict['preq'].head.repo.full_name
+    new_branch = job_obj.preq_dict['preq'].head.ref
+    # These are for the default app repo that goes with the workflow
+    auth_repo = job_obj.repo["app_address"]
+    auth_branch = job_obj.repo["app_branch"]
```

Comment:
Are these guaranteed to exist? Should the program fail if they don't?

Reply:
Some are from the github api results, and some are from the config file. They should exist, but something could go wrong, so no guarantees. A try block?

Reply:
I think that it's fine to fail where they don't exist. It will fail on a KeyError exception. We might consider using a default value, if none is found. I was thinking something like:

```
auth_branch = job_obj.repo.get("app_branch", "develop")

Reply:
If this doesn't work, there's a problem with the config file. I changed it to a try/except block on my laptop, and error out. I'll push it after a successful test.